### DISCUSSION TITLE
Suggestion for solving the "spear+rooftop" exploit.

### DISCUSSION THREAD

**Discussion by blot-runner** [cataclysmbn/Cataclysm-BN#5775] [General]
### Is your feature request related to a problem? Please describe.

As far as I know this is a known feature/exploit in the community, but I was unable to find any issues/PRs on it, so I'll describe it in some detail. As of now having a reach weapon and being able to climb to a roof trivializes many locations, including those that are supposed to be a challenge to a player (at least judging by the rewards). The strategy is as follows: equip a reach weapon, climb to a roof, attract zombies, wait till they approach, hold 'f' until everyone is dead, wait to regain stamina, rinse and repeat. Example: I've just cleared out a FEMA camp by a newly created char with mediocre combat abilities (8 str, 8 dex, 2 melee, 2 dodge, 2 piercing, no combat traits - in fact, no traits at all), several copper spears and a stepladder (both items are an easy craft even in the very beginning).

### Describe the solution you'd like

Expand the grab attack to also grab reach attack weapons: if a zombie that has a GRAB as a possible attack gets hit with a reach attack from a higher z-level, it has a chance to perform a counter (similar to the ZAPBACK functionality). Depending on how deadly you want it to be, the counter itself can be pulling the weapon out of the player's hands (similar to technician's PULL_METAL_WEAPON, but not restricted by material) or outright pulling the character off the roof (as with wrestler's RANGED_PULL). The former is a nuisance, but, IMO, still enough to clamp on the exploit, unless one brings scores of spears, while the latter is, in effect, a death sentence.
Reality-wise: zombies grabbing at the character should be just as happy to grab at any object shoved at them; grabbing a long spear is relatively easy; once grabbed and pulled forward, the spear owner can try to wrestle it free if on the level ground and not high above the opponent, but if caught on the ledge, their choices are basically to let go of the weapon or fall down.

**Comment by Soadreqm** [comment]
Generally, I feel like the rooftop hoplite and similar tricks _should_ work against most unintelligent enemies, but only most. There should be a few wildcards.

I think this suggestion has a couple problems, though. One, the grabbers would presumably counter spears even on level ground, and I'm not sure if that's desirable.

Two, the obvious counter to this is to selectively only stab the zombies that don't have grab attacks. Kill every non-grabber in the mob, and then mop up the grabbers on the ground. Thing is, that's considerably more keystrokes and attention required for something that requires zero decisionmaking. Like, suppose autoattack selectively targeted non-grabber enemies, and gave an "are you sure" prompt if there were only grabber enemies in reach. Would you ever want to answer "yes"? Would you ever want to manually target a grabber enemy if there were other targets? 

That wouldn't be a problem if grabber zombies could grab spears when a reach attack is made against a monster _near_ them, letting them cover a whole group of zombies. Communicating to the player that that can happen is also a bit of a UI problem, though. And it makes spears even more risky to use on level ground.

On a somewhat similar line, I've been thinking it'd be cool if the runner-type zombies could climb. The fast ones clamber up walls in Half-Life 2, it seems appropriate. That wouldn't be a hard counter to all spear weapons, but it would be another monster that can still attack you on the roof.

**Comment by scarf005** [comment]
i've said this a few times on discord but can't find it written here, so my idea is:

make tiles with large volumes (500L?) act as ramp, so naively spearing zombies will eventually gather up pile of corpses, which other zombies can use to climb into next floor.
this way while spearing from rooftop is viable, it won't be 100% safe

**Comment by blot-runner** [comment]
@Soadreqm 
> Generally, I feel like the rooftop hoplite and similar tricks _should_ work against most unintelligent enemies, but only most. There should be a few wildcards.

The problem I'm specifically referring to is the susceptibility of "high risk high reward" enemies to this tactics (soldies, burners, bio-operators). So, if this specific problem is to be addressed, these are the wildcards this tactics shouldn't work against.

> I think this suggestion has a couple problems, though. One, the grabbers would presumably counter spears even on level ground, and I'm not sure if that's desirable.

My suggestion specifically calls for one of the conditions to be "the attack is from a higher z-level".

> Two, the obvious counter to this is to selectively only stab the zombies that don't have grab attacks. Kill every non-grabber in the mob, and then mop up the grabbers on the ground. Thing is, that's considerably more keystrokes and attention required for something that requires zero decisionmaking. Like, suppose autoattack selectively targeted non-grabber enemies, and gave an "are you sure" prompt if there were only grabber enemies in reach. Would you ever want to answer "yes"? Would you ever want to manually target a grabber enemy if there were other targets?
>
> That wouldn't be a problem if grabber zombies could grab spears when a reach attack is made against a monster _near_ them, letting them cover a whole group of zombies. Communicating to the player that that can happen is also a bit of a UI problem, though. And it makes spears even more risky to use on level ground.

I'm sorry, I guess, my initial post was poorly worded. I wasn't talking about grabber zombie, I was talking about all the zombies with the GRAB attack. The one that allows the zombie to grab the character. To my knowledge, it's ALL the zombies in the game (and, as I've just discovered, some non-zombies too - not sure if this is a desirable outcome then a spider can do this, but don't they all climb anyway?).
 
> On a somewhat similar line, I've been thinking it'd be cool if the runner-type zombies could climb. The fast ones clamber up walls in Half-Life 2, it seems appropriate. That wouldn't be a hard counter to all spear weapons, but it would be another monster that can still attack you on the roof.

I agree it's a cool idea in principle (as T, Sean Collins put it, "If they could run, they could climb"), but since 2-3 runners cannot compare to 10+ soldiers, it will do nothing for the encounters like FEMA camp. At present it's a free looting ground, I've no idea if this is intended, but with the recent nerfs to the "dead soldiers" special it doesn't seem so.

@scarf005 
>i've said this a few times on discord but can't find it written here, so my idea is:
>
>make tiles with large volumes (500L?) act as ramp, so naively spearing zombies will eventually gather up pile of corpses, >which other zombies can use to climb into next floor.
>this way while spearing from rooftop is viable, it won't be 100% safe

But do the zombies pathfind that well? If you just go several tiles to the side, will they follow you on their level, or will they go away from you to ascend the new ramp and get you? I've never checked it, will try to later in the day, but maybe you do have an answer already?
EDIT: I've checked it and they don't. It's enough to move 2-3 tiles into any direction, and they happily walk over the ramp to stand on the ground below you and get slaughtered.

**Comment by ZChris13** [comment]
I believe the easiest solution to this is zombies GROUP_BASHing down the walls if you try to tab too large of a group. This has actually happened to me a few times when I was experimenting with gyrocopter raiding, and is why I don't do that anymore.

**Comment by blot-runner** [comment]
> I believe the easiest solution to this is zombies GROUP_BASHing down the walls if you try to tab too large of a group. This has actually happened to me a few times when I was experimenting with gyrocopter raiding, and is why I don't do that anymore.

First, looks like there is a bug that allows you to stand on thin air when the wall below is destroyed, please see #5766.
Second, even when it's fixed - many walls are too tough for anything short of hulk to tear down, regardless of the mob size.

**Comment by Soadreqm** [comment]
@blot-runner 
>I'm sorry, I guess, my initial post was poorly worded. I wasn't talking about grabber zombie, I was talking about all the zombies with the GRAB attack.

Ah. Yes, that's much more extensive. What enemies even are there that warrant climbing to a roof, but can't grab you?

I don't think I like this idea at all. As I said, I think the rooftop hoplite strat _should_ generally work. The main enemy in the game is zombies, which are generally depicted as unthinking lumbering husks, closer to machines than animals. _Clearly_ they should be vulnerable to an intelligent opponent with superior terrain. They should stand helplessly in your kill zone while you poke them to death with a stick. That's how zombies are supposed to work.

Also, if you feel that the rooftop hoplite strat _shouldn't_ work, the better solution would be to just disable reach attacks across z-levels. Seriously, what would it even be for, if every enemy worth fighting like that is immune?

**Comment by ZChris13** [comment]
does RANGED_PULL work across z-levels
how about ranged attacks (shocker, spitter, corrosive)
and reach attacks (hollow, juggernaut)

**Comment by blot-runner** [comment]
> Ah. Yes, that's much more extensive. What enemies even are there that warrant climbing to a roof, but can't grab you?

Dogs, for instance. Or mi-go. Or medical zombies, they don't grab you. Besides, let me emphasize - "grab a spear" counter shouldn't proc after every attack. Just often enough to keep the player on their toes.

> I don't think I like this idea at all. As I said, I think the rooftop hoplite strat _should_ generally work. The main enemy in the game is zombies, which are generally depicted as unthinking lumbering husks, closer to machines than animals. _Clearly_ they should be vulnerable to an intelligent opponent with superior terrain. They should stand helplessly in your kill zone while you poke them to death with a stick. That's how zombies are supposed to work.

You raise valid points, if we look at how the zombies are depicted in media, they're exactly that - mindless automatons, dangerous mainly because of their numbers. Well, I guess, there's also a "sneaky zombie" trope in the media, like a crawler hiding under a car to bite somebody's ankle - but that's beside the point, it's still more of a situational cunning than a real tactical aptitude.
However,
1. From the game balance standpoint (namely, its "risk-reward" part), right now "rooftop hoplite" is a strategy that allows for obtaining high-value loot with zero risk and almost zero effort (really, the only thing that stops you from using it literally from the start of the game with literally any character is that you need a wood saw - and even that can be bypassed if you find a roof with a downspout). To illustrate: I've just created a 8/8/8/8 skill-less character and cleared out another FEMA camp like this, this one was a two corpse pits one, the worst in terms of loot. The haul (listing only weapons): 6 9mm handguns (4 brand new), 130 assorted 9mm rounds, 1 shotgun (brand new), 17 12ga slugs, 3 grenades, RM88 battle rifle (brand new), 160 8x40mm caseless rounds to it, RM232 machine pistol, 105 5x50mm caseless rounds to it (55 flechettes and 50 penetrators). To reiterate - I'm not counting armor, tools, food, books, medicals etc., but there is a lot of it too.
Isn't it a bit too much of a reward for a riskless strategy?

2. From a narrative standpoint, in the very same media, the "devise a simple trap to slaughter mindless foes en masse" plan (regardless of whether those foes are zombies, swarm insects, non-AI robots etc.) usually fails (or even backfires) precisely because it's a surefire way of completing the task - hence, it doesn't make for an engaging story. Now, IRL you don't want and engaging story, you want the things to be as sure as possible, because IRL losing a fight is anything but fun. But if we're talking about a game - isn't the thrill is why we're playing it? The mental high of being presented with a sudden challenge and still besting our enemy - not because we had a surefire strategy beforehand, but because we were resourceful, versatile and crazy-prepared enough to adapt to the unexpected? Isn't it the whole point beyond such things as special zombies, ferals etc.?

> Also, if you feel that the rooftop hoplite strat _shouldn't_ work, the better solution would be to just disable reach attacks across z-levels. Seriously, what would it even be for, if every enemy worth fighting like that is immune?

I believe that if there is something worse than a broken mechanics, it's patching it with arbitrary "you are not allowed to do that" prohibitions not grounded in reality or logic. Moreover, I agree with you that such a strategy should be at least somewhat plausible. However, I think the game would benefit from making it less of a "hold "f" to win" thing. Take a similar strategy of a "rooftop archer" for example - it's also risk-free, but much less broken - you'll have to train Fabrication and Archery, craft enough arrows, maybe find a book for making a better bow. If you misjudge your aiming skills or don't make enough arrows, you might get stuck without an easy way out. Whereas, a "rooftop hoplite" - well, I think, by now I've made enough examples on how utterly broken this strategy is at present. If a zombie will have a decent chance of yanking a spear from a rooftop hoplite hands (not 100% - just decent enough to keep this in mind), the strategy still remains usable, but it will require more planning and care in its execution. _How large a horde I can take with the spears I have? Should I risk it or not?_ And so on. 
Maybe what I'm suggesting is an overkill - what are your thoughts on how to make this mechanics less broken? Or do you feel it's perfect as it is?

**Comment by blot-runner** [comment]
> does RANGED_PULL work across z-levels
> how about ranged attacks (shocker, spitter, corrosive)
> and reach attacks (hollow, juggernaut)

Yes, to all of these. However, these are all from the evolved zombies, they don't arrive until later in the game. By then you probably don't care about the rooftops at all, especially if you've got a good headstart in the loot and stats/skills training by killing the standard zombies from the roof.

**Comment by Soadreqm** [comment]
>"grab a spear" counter shouldn't proc after every attack. Just often enough to keep the player on their toes.

But there's no real way for the player to respond to it. It's not like getting disarmed by a zombie technician. You can't switch to the equivalent of a non-magnetic weapon, since the only item in the game with both REACH_ATTACK and NO_UNWIELD is the monofilament whip, and you probably don't have that in the early game. You can't prioritize killing the enemies with the special ability, since _basic zombies_ have it. 

It's also not like having a weapon break on you, since you can avoid that by getting a better weapon. A simple knife spear might already be destroyed over the course of a prolonged roof stabbing session, but a lucerne hammer probably won't. If zombies can disarm both from you, then they're equally disposable.

>Dogs, for instance. Or mi-go. Or medical zombies, they don't grab you.

How often does fighting dogs or medical zombies require that you climb to a roof? Like, dogs are ubiquitous, but they appear among other zombies. If you want to engage them from a roof, you'd need to manually aim every single attack. Or rewrite the auto-aim logic for reach attacks, but I somehow doubt that'd be trivial. As for mi-go, I feel that it _shouldn't_ really work against them, even if it currently does. Intelligent enemies should ideally try to avoid the player's attack range, if they clearly have no way to attack you back.

>I believe that if there is something worse than a broken mechanics, it's patching it with arbitrary "you are not allowed to do that" prohibitions not grounded in reality or logic.

I'm having a hard time seeing how your suggestion is grounded in reality. Poking at zombies from a roof is too effective, so zombies are now able to disarm you, but _only_ if you are performing a reach attack and _only_ if it's from a different z-level. I mean, it's not completely unreasonable to lose hold of your weapon because an enemy grabbed it, but that doesn't happen in any other circumstance, and you've explicitly said that it _shouldn't._

I get what you're saying about risk/reward, but I think it would be better to try to balance it in some other way first. Like putting stronger enemies in places where the loot is currently too easy. Do you think you could still take the FEMA camp day one if there were guaranteed to be a dozen basic ferals there? The thrown rocks don't hurt a lot, but they _do_ hurt, and they easily outrange your spear.

**Comment by ZChris13** [comment]
if you feel like the risk/reward for certain areas is off due to being able to combine beating the evolution clock with roof hoplite, then may I simply suggest suggesting:

buffing those areas with early spitters or a small pile of ferals
nerfing the buildings in the area so they're vulnerable to being torn down by basic zombies

FEMA shelters having full on concrete buildings was always a little bit weird to me anyway

**Comment by blot-runner** [comment]
@Soadreqm 

> But there's no real way for the player to respond to it. It's not like getting disarmed by a zombie technician. You can't switch to the equivalent of a non-magnetic weapon, since the only item in the game with both REACH_ATTACK and NO_UNWIELD is the monofilament whip, and you probably don't have that in the early game. You can't prioritize killing the enemies with the special ability, since _basic zombies_ have it.
> 
> It's also not like having a weapon break on you, since you can avoid that by getting a better weapon. A simple knife spear might already be destroyed over the course of a prolonged roof stabbing session, but a lucerne hammer probably won't. If zombies can disarm both from you, then they're equally disposable.

I'm sorry, it's the same as saying there is no way for a player to respond to a situation of shooting the zombies from the ground then realizing they're drawing more living dead than they have ammo. In both cases there is a finite resource and an uncertainty on if it will last for long enough, in both cases one is able to hedge by packing more of that resource, in both cases having an exit strategy is vital, since once in a while you'll misjudge your abilities. What makes the "rooftop hoplite" strategy so special that it should be allowed to not require the player to have a contingency plan or exit strategy?


> How often does fighting dogs or medical zombies require that you climb to a roof?

Funny, this is exactly how I ended up on one during the first town incursion in my current run (had no spear or desire to be a hoplite, so just climbed down the downspout the other side of the roof and fled). But that's beside the point, please see below.

> Like, dogs are ubiquitous, but they appear among other zombies. If you want to engage them from a roof, you'd need to manually aim every single attack. Or rewrite the auto-aim logic for reach attacks, but I somehow doubt that'd be trivial.

So, in that particular case the tactics will require the player to actually AIM for the enemies it can be safely used against, instead of just holding 'f' until everyone is dead. Like, actually PAYING ATTENTION to the fight. Did I get you right that you see it as a BAD thing?

> As for mi-go, I feel that it _shouldn't_ really work against them, even if it currently does. Intelligent enemies should ideally try to avoid the player's attack range, if they clearly have no way to attack you back.

In principle, I agree with you, but I fail to see how this particular point is relevant for the present discussion. You asked if there are enemies that at present could still be cheesed even if my suggestion is acted upon. Turns out, there are some. I never argued that there will still exist a meaningful population of enemies that are supposed to be a credible threat to a new character and still be cheesed under my suggestion - in fact, this is exactly what it is aimed to fix.


> I'm having a hard time seeing how your suggestion is grounded in reality.

Let me copy-paste the last paragraph of the initial post:

"Reality-wise: zombies grabbing at the character should be just as happy to grab at any object shoved at them; grabbing a long spear is relatively easy; once grabbed and pulled forward, the spear owner can try to wrestle it free if on the level ground and not high above the opponent, but if caught on the ledge, their choices are basically to let go of the weapon or fall down."


>Poking at zombies from a roof is too effective, so zombies are now able to disarm you, but _only_ if you are performing a reach attack and _only_ if it's from a different z-level. I mean, it's not completely unreasonable to lose hold of your weapon because an enemy grabbed it, but that doesn't happen in any other circumstance, and you've explicitly said that it _shouldn't._

We're talking very special circumstances here: First, the character is holding a long and rather heavy weapon and the grabber has the gravity advantage. Second, on level ground the character can afford to stumble to keep hold of a weapon, while on the edge of the roof it might spell a death for them, so they'll be much more inclined to sacrifice a weapon, and not their life. Third, the character is stabbing downwards at an acute angle, which means that there is no way they can hold the shaft close to their center of mass; the vector of the force applied to the spear by a grabber goes along the shaft, so, between this and the fact that the character's center of mass is some height above the ground the grabber has a huge mechanical advantage on their side.


> I get what you're saying about risk/reward, but I think it would be better to try to balance it in some other way first. Like putting stronger enemies in places where the loot is currently too easy. Do you think you could still take the FEMA camp day one if there were guaranteed to be a dozen basic ferals there? The thrown rocks don't hurt a lot, but they _do_ hurt, and they easily outrange your spear.

Putting ferals/spitters into FEMA camps is a passable stopgap solution, and it does solve the most glaring instance of the problem, but the issue is bigger than that. And I really think that the main problem in our discussion is that we don't agree if there is a fundamental issue in the first place. Let me formulate the points that led me to making the present suggestion:

1. At present the "rooftop hoplite" tactics can be implemented almost as soon as the game starts.
2.  Said tactics allow the player to kill more or less any amount of the zombies encountered in the initial phase of the game (before there is a notable chance of encountering those evolved enough to get you on the roof) with zero risk.
3. Said zombies are supposed to be a credible threat to a new character.
4. Having an easily implemented tactics that allows for riskless and effortless slaughter of infinite (for practical purposes) amount of the enemies is not compatible to these enemies being a credible threat.
5. Ergo: changing the present state of events so that at least one of the first three points is no longer true is desirable.

My reasoning:
Point 1 - follows from the crafting recipes and the loot and furniture readily present in the common locations (initially I thought wood sawing 1 is the only thing that can pass as a gatekeeper in a very bad light - now I've realized primitive adze is enough, making the setup literally available right from the start in all the scenarios sans specifically confined ones like an multi-storey rooftop or a lab that don't have much room for this tactics anyway.
Point 2 - verified personally, having consistently failed to find a scenario that allows for any way for the character to get into the harm's way while killing scores of zombies by doing nothing but holding 'f' and resting/moving occasionally. This tactics also seems to be a well-known one within the community and the only counter to it I ever heard other players mentioning is climbers, ranged attacks or RANGED_PULL - all not present in the starting zombie population.
Point 3 - indirectly confirmed by devs in #5629 and this Reddit post https://www.reddit.com/r/cataclysmbn/comments/1gflr00/psa_game_difficulty_tuning_now_happening_on/
Point 4 - logical conclusion of the first three points (from the definition of threat as 'something that can harm' and 'riskless' as 'having no possibility to induce harm').
Point 5 - logical conclusion of 4 and an implication that making the game better is desirable.

Now, could you please state which of the points above you disagree with? In addition, I'm sorry, but it seems that I didn't get your answer to my previous question: **Maybe what I'm suggesting is an overkill - what are your thoughts on how to make this mechanics less broken? Or do you feel it's perfect as it is?**

@ZChris13 

> if you feel like the risk/reward for certain areas is off due to being able to combine beating the evolution clock with roof hoplite, then may I simply suggest suggesting:
> 
> buffing those areas with early spitters or a small pile of ferals nerfing the buildings in the area so they're vulnerable to being torn down by basic zombies
> 
> FEMA shelters having full on concrete buildings was always a little bit weird to me anyway

I suspect the sturdy walls were implemented as a way to make the player hunt for the ID cards to get to the best loot. From a realism standpoint, it all depends on how permanent this particular camp is (after all, concrete is not that hard to pour and I'd assume one would want the valuable stuff to be protected by more than a layer of plaster over wood and a warning sign). Otherwise, please refer to my reasoning above in the discussion with Soadreqm.

**Comment by scarf005** [comment]
the discussion is getting pretty lengthy - let's continue this on actual discussion.

**Reply by ZChris13** [reply]
I don't foresee this going anywhere productive

**Comment by Soadreqm** [comment]
>I'm sorry, it's the same as saying there is no way for a player to respond to a situation of shooting the zombies from the ground then realizing they're drawing more living dead than they have ammo. In both cases there is a finite resource and an uncertainty on if it will last for long enough, in both cases one is able to hedge by packing more of that resource, in both cases having an exit strategy is vital, since once in a while you'll misjudge your abilities.

Is it really? You know exactly how many bullets you have, but being disarmed would presumably be a random chance. I don't think you've mentioned what kind of random chance it would be, whether it's one in five, one in a hundred or one in a thousand, but regardless, you can't actually know how many attacks you can perform with your spear before it's gone.

You can also carry a LOT of ammo, while reach weapons tend to be quite large. The whip and the scourge are 1L, but they're both fabrication 6 and rare to find.

I just don't see how this would keep anyone "on their toes". You have to attack several times to kill even the least zombie, and every attack would have a nonzero chance to instantly destroy your weapon. Well, not destroy. Make impossible to recover until you win the fight. That's an awful gamble. Either you spend a week and a thousand units of leather preparing, and bring a duffle bag full of scourges, or you just avoid the fight.

And isn't that your stated goal? To make players avoid fighting enemies from rooftops with reach weapons. It seems a bit disingenuous to present it as a tactic on par with _"use a gun"_ when you clearly don't want it to be worth doing.

>So, in that particular case the tactics will require the player to actually AIM for the enemies it can be safely used against, instead of just holding 'f' until everyone is dead. Like, actually PAYING ATTENTION to the fight. Did I get you right that you see it as a BAD thing?

Yes. I said as much in my first reply, when I thought you just wanted to upgrade the grabber line of zombies instead of all of them. If a task requires no active decisionmaking, then it's generally a bad thing for it to require a lot of keystrokes and constant attention. I wasn't expecting this to be a controversial statement.

For a similar case, trying to smash an acidic corpse currently prompts to check if you're sure. You're not expected to pay attention to the task of pulping corpses, or to check each tile before smashing it. The game just does it for you.

>I agree with you [regarding the mi-go], but I fail to see how this particular point is relevant for the present discussion.

It isn't really, just thought I'd mention it since it came up.

I intend to reply to the rest of the post, but this is getting _LONG_ and I need a break.

**Comment by Soadreqm** [comment]
_The realism argument._ I want to open by saying that none of this is a deal breaker for me. It's a video game, sometimes you might need to make arbitrary balance changes. It is what it is. But you're trying to argue that your proposed changes are grounded in reality, which they _aren't._

>Reality-wise: zombies grabbing at the character should be just as happy to grab at any object shoved at them; grabbing a long spear is relatively easy; 

It really isn't. I mean I'm no martial arts expert, and if you know any HEMA youtubers who've said that grabbing a spear is actually really easy and being grabbed was the number one cause of death for 15th century pike formations, then feel to link them and and prove me wrong. But nothing about it _looks_ easy. You're talking about a relatively small, moving target that someone else is actively trying to keep you from grabbing. It requires fine motor control and reaction speed. Conversely, to grab a human, a zombie just needs to _hug_ them.

>once grabbed and pulled forward, the spear owner can try to wrestle it free if on the level ground and not high above the opponent, but if caught on the ledge, their choices are basically to let go of the weapon or fall down.

This whole line of argumentation hinges on the assumption that the zombie grabbing the spear by the sharp end is able to exert more control over it than the human grabbing it by the handle. Now, there _are_ possible arguments you could make for why that would be the case. Maybe the blob makes the zombies really sticky. Maybe they all just have amazing grip strength. But you never make them, which is just as well since they aren't compatible with your justification for why zombies wouldn't be able to do it on level ground.

>We're talking very special circumstances here: First, the character is holding a long and rather heavy weapon and the grabber has the gravity advantage.

1.45 kg for the copper spear. They're mostly in that range. A pike is somewhat heavier at 2.5 kg, and the war scythe is something of an outlier at 3.013 kg. That's heavy for a weapon, but the "gravity advantage" really isn't that much. Fifteen Newtons extra, if you're pulling straight up? You don't need to lift the zombie, just pull the spear from it's grasp. If you're unable to, well, then it wouldn't really help if you were on the same z-level, would it?

>Second, on level ground the character can afford to stumble to keep hold of a weapon, while on the edge of the roof it might spell a death for them, so they'll be much more inclined to sacrifice a weapon, and not their life. 

If you're unable to maintain hold of your spear by pulling back on it, what advantage does stepping _forwards_ give you? Not that you'd have to actually step forward. By your proposal, disarms would just _never happen_ if you were on the same z-level with the zombie. The implied grab break is a free action and you can do it even if there's a spiked pit right in front of you.

>Third, the character is stabbing downwards at an acute angle, which means that there is no way they can hold the shaft close to their center of mass; the vector of the force applied to the spear by a grabber goes along the shaft, so, between this and the fact that the character's center of mass is some height above the ground the grabber has a huge mechanical advantage on their side.

First off, the grabber also can't hold the shaft close to their center of mass, on account of you're stabbing his head and that's where he needs to grab the spear if he wants to hold it at all. Also, the player's bracing with their legs against the lip of the roof, which is a fairly stable position for lifting things, so I don't really see where the zombie would get a mechanical advantage. 

Leaving that aside, there's a number of things you would realistically be able to do to make your spear harder to grab.

- Make the forward part of the shaft more slippery. Technology available to the player ranges from rubbing butter on it to applying a non-stick diamond coating via carbon vapor deposition. 
- Add a sacrificial tear-away grip surface. For some reason, historical spears usually don't seem to have had these, but it seems like it'd be easy to do. Tape or tie a few layers of plastic bags, newspapers or bark to the shaft. The zombie tries to grab it, and you just pull the spear away, leaving the zombie clutching at trash. Apply a fresh spear condom and resume.
- Add a lanyard. You securely attach a loop to the end of the spear, and tie a rope to it. The other end of the rope you tie to a chimney. The spear is now impossible to pull down from the roof - even if the zombies pull it from your hands, you can just pull it back with the rope. Worked for whalers.

**Reply by RosieMcCattail** [reply]
>     * Add a sacrificial tear-away grip surface. For some reason, historical spears usually don't seem to have had these, but it seems like it'd be easy to do. Tape or tie a few layers of plastic bags, newspapers or bark to the shaft. The zombie tries to grab it, and you just pull the spear away, leaving the zombie clutching at trash. Apply a fresh spear condom and resume.

Historically, if a human tries to grab a spear and fails, he gets stabbed, and dies, which is a bad thing, so he doesn't grab it at all.
Historically, zombies don't really care about pain much (yes, I know zombies flinch and let you go when you hit them, I don't know how this ties to zombies fighting you to the death), so they don't care whether they will succeed or not, so they grab it. Or they try it.

However, and most importantly, historically, spears are not used by hovering them 2 cm away from the target while the spearman is looking for a sweet spot to poke, they're used by having them at ready near your body and then shoving the whole thing forward, because, you know, the more speed - the more damage and that stuff, so the potential grabber has just a split second to grab the spear which will that very instant be pulled back to the spear user, most likely tripping the graber over. Try grabbing a piston on its forward motion - you'll be lucky to keep your hands as it's retracting back. Now, zombies don't care about keeping hands, so they still would try, but the whole concept of grabbing a rapidly moving object midair is too dubious for a brainless piece of blob-animated flesh.

**Comment by ZChris13** [comment]
this, in general, seems like an attempt to inject anti-fun into the game in an attempt to solve something that could be better addressed with map design

**Comment by Soadreqm** [comment]
>Putting ferals/spitters into FEMA camps is a passable stopgap solution, and it does solve the most glaring instance of the problem, but the issue is bigger than that.

Is it really? I think your reasoning for why there _is_ an issue is sound, but it does not follow that the issue couldn't be solved just by tweaking enemy spawns. You can make the player unable to kill infinite zombies without getting hurt, without taking away the player's ability to kill zombies. Ferals especially seem good for this. They can threaten you at range, but aren't particularly dangerous if you fight them in melee. New enemies could be added too. Maybe something like the headcrab tosser from Half-Life 2. We don't have headcrabs, but it could be a zombified crazy cat lady or something like that.

I don't even think you need to raise the threat level all that much to make roof stabbing a lot less chill. If most of the horde can't touch you, but you're being attacked enough that you can't rest to recover stamina, that's still enough to put you in mortal danger if you don't have a backup plan.

>Maybe what I'm suggesting is an overkill - what are your thoughts on how to make this mechanics less broken?

Besides adding enemies with ranged attacks, or enemies that climb, I think climbing up a waterspout should maybe be a bit less reliable. As in take a bit of time, have a chance to fail if you're attacked while climbing. That's not going to matter if you're always carrying a ladder everywhere you go, but most people don't.

It's also possible to just tweak individual locations if they have amazing loot and are too easy to clear. For the FEMA camp specifically, you could place all the concrete structures underground. It'd still stop you from getting in without the keycard, but you wouldn't have concrete walls blocking zombie movement.

**Reply by ZChris13** [reply]
moving FEMA concrete structures underground is a funny idea, I like that
